### PR TITLE
[feat](nereids)set runtime filter wait time according to table row count and table type

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
adjust compute rf wait time according to max table row count, if wait time is not default value
- olap:
  -- row < 1G: 1 sec
  -- 1G <= row < 10G: 5 sec
  -- 10G < row: 20 sec
- external:
  -- row < 1G: 5 sec
  -- 1G <= row < 10G: 10 sec
  -- 10G < row: 50 sec
     
Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by morrySnow** [comment]
add desc please

File: fe/.idea/vcs.xml
Comment (morrySnow) [inline fe/.idea/vcs.xml:1]: revert this

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java [hunk 1]
```diff
@@ -279,6 +281,39 @@ private Plan planWithoutLock(
             disableJoinReorderReason.ifPresent(statementContext::setDisableJoinReorderReason);
         }
 
+        // compute rf wait time according to max table row count, if wait time is not default value
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:284]: better to wrap it into a function, to make the whole planner process clean

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -1136,7 +1136,7 @@ public class SessionVariable implements Serializable, Writable {
     private int runtimeBloomFilterMaxSize = 16777216;
 
     @VariableMgr.VarAttr(name = RUNTIME_FILTER_WAIT_TIME_MS, needForward = true)
-    private int runtimeFilterWaitTimeMs = 1000;
+    public int runtimeFilterWaitTimeMs = 1000;
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1139]: why change this but use the setRuntimeFilterWaitTimeMs instead?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java [hunk 3]
```diff
@@ -228,6 +230,44 @@ public Plan planWithLock(LogicalPlan plan, PhysicalProperties requireProperties,
         }
     }
 
+    /**
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:233]: 1. better to add this into pr desc
2. double-check the basic benchmark such as tpcds 1T olap/external table performance, after removing the config.

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run cloud_p0

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run external

**Comment by englefly** [comment]
run cloud_p0

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**